### ISSUE TITLE
7900XTX Sensor Support

### ISSUE THREAD

**Issue by LeagueRaINi** [CXWorld/CapFrameX#234]
CapFrameX doesnt seem to detect my 7900xtx properly and doesnt show any sensor data, ~~tried to get the src running locally to debug but so far i've not been very successful in doing so~~ in the meantime i'm going to throw up this issue

<details>
  <summary>Gpu Info</summary>

![raw](https://user-images.githubusercontent.com/19365835/213868569-2aa4ccac-6dbc-4bef-91d8-9ad8c10d6d47.png)
</details>

<details>
  <summary>CapFrameX.log</summary>

```json
{"@t":"2023-01-21T13:10:31.5026583Z","@mt":"CapFrameX {version} started at UTC {atomicTime}","version":"1.7.0.4","atomicTime":"13:10:31.2360000","SourceContext":"Microsoft.Extensions.Logging.ILogger"}
{"@t":"2023-01-21T13:10:31.5537174Z","@mt":"Window dimensions are {width} x {height}. Window position is {positionLeft} x {positionTop}","width":1498,"height":800,"positionLeft":-1,"positionTop":45,"SourceContext":"Microsoft.Extensions.Logging.ILogger"}
{"@t":"2023-01-21T13:10:31.6097169Z","@mt":"{componentName} Ready","@l":"Debug","componentName":"SensorService","SourceContext":"CapFrameX.Sensor.SensorService"}
{"@t":"2023-01-21T13:10:31.6097169Z","@mt":"SensorService {initializationTime}s initialization time","initializationTime":0,"SourceContext":"CapFrameX.Sensor.SensorService"}
{"@t":"2023-01-21T13:10:31.6597148Z","@mt":"Start creating CPU group."}
{"@t":"2023-01-21T13:10:31.6797144Z","@mt":"CPUID vendor info: {vendor}.","vendor":"AMD"}
{"@t":"2023-01-21T13:10:31.6797144Z","@mt":"CPUID family code: {family}.","family":"25"}
{"@t":"2023-01-21T13:10:31.6877136Z","@mt":"CPUID core count: {coreCount}.","coreCount":12}
{"@t":"2023-01-21T13:10:31.6877136Z","@mt":"CPUID thread count per core: {coreThreadCount}.","coreThreadCount":2}
{"@t":"2023-01-21T13:10:31.7107181Z","@mt":"LoginManager {initializationTime}s initialization time","initializationTime":0,"SourceContext":"CapFrameX.Data.LoginManager"}
{"@t":"2023-01-21T13:10:31.7937158Z","@mt":"Now observing directory: {path}","path":"C:\\Users\\RaINi\\Documents\\CapFrameX\\Captures","SourceContext":"CapFrameX.Data.RecordDirectoryObserver"}
{"@t":"2023-01-21T13:10:31.9797171Z","@mt":"{componentName} Ready","@l":"Debug","componentName":"OverlayEntryProvider","SourceContext":"CapFrameX.Overlay.OverlayEntryProvider"}
{"@t":"2023-01-21T13:10:31.9877179Z","@mt":"{componentName} Ready","@l":"Debug","componentName":"OverlayService","SourceContext":"CapFrameX.Overlay.OverlayService"}
{"@t":"2023-01-21T13:10:31.9927174Z","@mt":"OverlayService {initializationTime}s initialization time","initializationTime":0,"SourceContext":"CapFrameX.Overlay.OverlayService"}
{"@t":"2023-01-21T13:10:32.0117160Z","@mt":"{viewName} Ready","@l":"Debug","viewName":"CaptureViewModel","SourceContext":"CapFrameX.ViewModel.CaptureViewModel"}
{"@t":"2023-01-21T13:10:32.0497159Z","@mt":"PresentMon successfully started","SourceContext":"CapFrameX.PresentMonInterface.PresentMonCaptureService"}
{"@t":"2023-01-21T13:10:32.5357686Z","@mt":"ReportViewModel {initializationTime}s initialization time","initializationTime":0,"SourceContext":"CapFrameX.ViewModel.ReportViewModel"}
{"@t":"2023-01-21T13:10:32.6537658Z","@mt":"WebServer (http://*:1337) State - Loading"}
{"@t":"2023-01-21T13:10:32.6557658Z","@mt":"WebServer (http://*:1337) State - Listening"}
{"@t":"2023-01-21T13:10:36.2547067Z","@mt":"Error when starting OpenHardwareMonitor","@l":"Error","@x":"System.InvalidOperationException: Nullable object must have a value.\r\n   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGPU.Update()\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGPU..ctor(String name, Int32 adapterIndex, Int32 busNumber, Int32 deviceNumber, IntPtr context, ISettings settings, ISensorConfig config, IProcessService processService)\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGroup..ctor(ISettings settings, ISensorConfig sensorConfig, IProcessService processService)\r\n   at OpenHardwareMonitor.Hardware.Computer.set_GPUEnabled(Boolean value)\r\n   at CapFrameX.Sensor.SensorService.<StartOpenHardwareMonitor>b__39_0() in D:\\Jenkins\\workspace\\CX-App_release_1.7.0\\source\\CapFrameX.Sensor\\SensorService.cs:line 182","SourceContext":"CapFrameX.Sensor.SensorService"}
{"@t":"2023-01-21T13:10:36.7877224Z","@mt":"Set overlay defaults","SourceContext":"CapFrameX.Overlay.OverlayEntryProvider"}
{"@t":"2023-01-21T13:10:36.7877224Z","@mt":"CPU detected: {cpuName}.","cpuName":"AMD Ryzen 9 5900X","SourceContext":"CapFrameX.Overlay.OverlayEntryProvider"}
{"@t":"2023-01-21T13:10:36.7877224Z","@mt":"CPU threads detected: {threadCount}.","threadCount":24,"SourceContext":"CapFrameX.Overlay.OverlayEntryProvider"}
{"@t":"2023-01-21T13:10:36.7887220Z","@mt":"GPU detected: {gpuName}.","gpuName":"Unknown","SourceContext":"CapFrameX.Overlay.OverlayEntryProvider"}
```
</details>

Seems to throw an exception in the OpenHardwareMonitor lib in `Update()`

```json
{
    "@t": "2023-01-21T13:10:36.2547067Z",
    "@mt": "Error when starting OpenHardwareMonitor",
    "@l": "Error",
    "@x": "System.InvalidOperationException: Nullable object must have a value.\r\n   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGPU.Update()\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGPU..ctor(String name, Int32 adapterIndex, Int32 busNumber, Int32 deviceNumber, IntPtr context, ISettings settings, ISensorConfig config, IProcessService processService)\r\n   at OpenHardwareMonitor.Hardware.ATI.ATIGroup..ctor(ISettings settings, ISensorConfig sensorConfig, IProcessService processService)\r\n   at OpenHardwareMonitor.Hardware.Computer.set_GPUEnabled(Boolean value)\r\n   at CapFrameX.Sensor.SensorService.<StartOpenHardwareMonitor>b__39_0() in D:\\Jenkins\\workspace\\CX-App_release_1.7.0\\source\\CapFrameX.Sensor\\SensorService.cs:line 182",
    "SourceContext": "CapFrameX.Sensor.SensorService"
}
```

// Edit:
trying to access `powerTotal.Value.Value` is whats throwing, seems like quite a few other temperature related variables are also just `null`
`GetValueOrDefault()` can be used to get around the exception
![raw](https://user-images.githubusercontent.com/19365835/213873034-548fcd60-cb03-4d18-ae79-16e90bba1a0e.png)

regarding a lot of the other sensors also returning null, these are the ones i could get with a cli test app by iterating `hardware.Sensors`, not sure how many i'am *supposed* to be able to get
```
Type: Temperature, Name: GPU Core,              Value: 28
Type: Temperature, Name: GPU Hot Spot,          Value: 33
Type: Fan,         Name: GPU Fan,               Value: 994
Type: Clock,       Name: GPU Core,              Value: 1437
Type: Clock,       Name: GPU Memory,            Value: 909
Type: Voltage,     Name: GPU Core,              Value: 0,721
Type: Load,        Name: GPU Core,              Value: 21
Type: Load,        Name: GPU Memory Controller, Value: 0
Type: Control,     Name: GPU Fan,               Value: 23
```

**Comment by Taxxor90** [comment]
Which version are you on? The latest [1.7.1 beta](https://github.com/CXWorld/CapFrameX/releases/tag/v1.7.1beta) should've already fixed that

**Comment by LeagueRaINi** [comment]
> 1.7.0 doesn't have any RDNA3 support at all as it was released before the GPUs came out.
> 
> The latest [1.7.1 beta](https://github.com/CXWorld/CapFrameX/releases/tag/v1.7.1beta) does.

good to know, either way the core issue doesnt seem to be fixed even in the 1.7.1 branch, it still does not check if the value is null before trying to access its underlying value

**Comment by Taxxor90** [comment]
> either way the core issue doesnt seem to be fixed even in the 1.7.1 branch, it still does not check if the variable is null before trying to access its underlying value

`else if (powerTotal.Value != null)
					{
						powerTotalValue = powerTotal.Value.Value;
					}`

seems right to me

**Comment by LeagueRaINi** [comment]
oh huh not was i was looking for when i briefly looked at the snippet so i completely missed it, nice!
